backend/service/service_proposal.go

```diff
@@ -48,25 +63,240 @@ func (service *ProposalService) QueryProposalsByHeight(height int64) []model.Pro
+		}
+
+		tmp := model.ProposalNewStyle{
+			ProposalId:           propo.ProposalId,
+			Title:                propo.Title,
+			Type:                 propo.Type,
+			Status:               propo.Status,
+			Votes:                tmpVoteArr,
+			VotingPowerForHeight: powerForHeigtMap[propo.VotingStartHeight].totalVotingPower,
+		}
+
+		l := model.Level{}
+		name, err := lcd.GetProposalLevelByType(propo.Type)
+		if err != nil {
+			logger.Error("get proposal level by type", logger.String("err", err.Error()), logger.String("param", propo.Type))
+		}
+		l.Name = name
+		if propo.Status == document.ProposalStatusDeposit {
+			coinAsDoc, err := lcd.GetMinDepositByProposalType(propo.Type)
+			if err != nil {
+				logger.Error("get min deposit", logger.String("err", err.Error()), logger.String("param", propo.Type))
+			}
+			l.GovParam = model.GovParam{
+				MinDeposit: model.Coin{
+					Amount: coinAsDoc.Amount,
+					Denom:  coinAsDoc.Denom,
+				},
+			}
+
+			tmp.InitialDeposit = depositInitmap[tmp.ProposalId]
+
+			if len(propo.TotalDeposit) > 0 {
+				tmp.TotalDeposit.Amount = propo.TotalDeposit[0].Amount
+				tmp.TotalDeposit.Denom = propo.TotalDeposit[0].Denom
+			}
+		}
+
+		if propo.Status == document.ProposalStatusVoting {
+
+			threshold, participation, err := lcd.GetThresholdAndParticipationMinDeposit(propo.Type)
+			if err != nil {
+				logger.Error("GetThresholdAndParticipationMinDeposit", logger.String("err", err.Error()), logger.String("param", propo.Type))
+			}
+			l.GovParam = model.GovParam{
+				Threshold:     threshold,
+				Participation: participation,
+			}
+		}
+
+		tmp.Level = l
+
+		proposals = append(proposals, tmp)
+
+	}
+
+	return proposals
+}
+
 func (service *ProposalService) QueryList(page, size int) (resp model.PageVo) {
 	var data []document.Proposal
 	sort := desc(document.Proposal_Field_SubmitTime)
-	if cnt, err := pageQuery(document.CollectionNmProposal, nil, nil, sort, page, size, &data); err == nil {
-		var proposals []model.Proposal
+	selector := bson.M{
+		document.Proposal_Field_ProposalId:        1,
+		document.Proposal_Field_Title:             1,
+		document.Proposal_Field_Type:              1,
+		document.Proposal_Field_Status:            1,
+		document.Proposal_Field_SubmitTime:        1,
+		document.Proposal_Field_DepositEndTime:    1,
+		document.Proposal_Field_VotingEndTime:     1,
+		document.Proposal_Field_Votes:             1,
+		document.Proposal_Field_VotingStartHeight: 1,
+		document.Proposal_Field_Final_Votes:       1,
+	}
+
+	if cnt, err := pageQuery(document.CollectionNmProposal, selector, nil, sort, page, size, &data); err == nil {
+		proposals := make([]model.ProposalNewStyle, 0, len(data))
+		heightArr := make([]int64, 0, len(data))
+		voterAddrArr := make([]string, 0, len(data))
+
+		for _, v := range data {
+			if v.Status == document.ProposalStatusVoting {
+				heightArr = append(heightArr, v.VotingStartHeight)
+				for _, addr := range v.Votes {
```

Comment:
delete remove duplicated code. voter addr is distinct in one proposal

Reply:
[how to delete duplicates in a slice](https://stackoverflow.com/questions/26172673/how-to-delete-duplicates-of-an-item-from-a-slice)